server/database.go

```diff
@@ -162,6 +163,44 @@ func (s *Server) registerDatabaseRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to patch database ID: %v", id)).SetInternal(err)
 		}
 
+		if databasePatch.ProjectID != nil {
+			toProject, err := s.composeProjectByID(ctx, *databasePatch.ProjectID)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Project ID not found: %d", *databasePatch.ProjectID))
+			}
+			if toProject.TenantMode == api.TenantModeTenant {
+				// For database being transferred to a tenant mode project, its schema version and schema has to match a peer tenant database.
+				// When a peer tenant database doesn't exist, we will return an error if there are databases in the project with the same name.
+				peerSchemaVersion, peerSchema, err := s.getSchemaFromPeerTenantDatabase(ctx, database.Instance, *databasePatch.ProjectID, database.Name)
+				if err != nil {
+					return err
+				}
+
+				if peerSchemaVersion != "" || peerSchema != "" {
```

Comment:
This check seems to be a bit odd. I am thinking either we don't have this check or we have the check separately?

if peerSchemaVersion != "" {
}

if peerSchema != "" {
}